tests/unit/tools/core/test_command_builders.py

```
@@ -20,6 +20,49 @@
+    """
+    names = {expected_names} if isinstance(expected_names, str) else expected_names
+
+    def which_side_effect(
+        name: str,
+        path: str | None = None,
+    ) -> str | None:
+        if path is not None:
+            # Venv scripts lookup: validate name and path
+            if name not in names:
+                return None
+            if not (path.endswith("/bin") or path.endswith("\\Scripts")):
```

Ruff (0.15.7)

[warning] Call `endswith` once with a `tuple`

Merge into a single `endswith` call

(PIE810)